### PR TITLE
Restrooms from backend

### PR DESCRIPTION
**Pull request by aiecel**
HI! Made that restrooms are now stored in the backend! Any critique welcomed!!!!!

### CODE REVIEW

File: crap-advisor-frontend/src/core/components/Map/index.css [hunk 1]
```diff
@@ -1,5 +1,4 @@
 .map {
-    position: fixed;
-    top: 0;
-    left: 0;
+    width: 100%;
+    height: 100%;
```
Comment (akichula) [inline crap-advisor-frontend/src/core/components/Map/index.css:3]: Perhaps you should add
`max-width: 100%; 
max-height: 100%; 
min-width: 100%; 
min-height: 100% `
 to the correct display on some devices

File: crap-advisor-frontend/src/core/components/Map/index.jsx [hunk 2]
```diff
@@ -1,29 +1,39 @@
...
-import {YMaps, Map as YMap, ZoomControl, Placemark} from '@pbe/react-yandex-maps';
+import React, {useEffect, useState} from "react";
 import "./index.css"
-import restroomImage from '../../../assets/restroom.png'
+import {MapContainer, TileLayer, ZoomControl} from "react-leaflet";
+import RestroomMarker from "../RestroomMarker";
+
+const apiUrl = "http://localhost:8082"
+const defaultPosition = [53.21176, 50.18394]
+const defaultZoom = 13
```
Comment (akichula) [inline crap-advisor-frontend/src/core/components/Map/index.jsx:8]: Keep all of your variables inside a component

File: crap-advisor-frontend/src/core/components/Map/index.jsx [hunk 3]
```diff
@@ -1,29 +1,39 @@
...
-                    options={{
-                        iconLayout: "default#image",
-                        iconImageHref: restroomImage,
-                        iconImageSize: [32, 32],
-                        iconContentOffset: [32, 32]
-                    }}
-                />
-            </YMap>
-        </YMaps>
+        <MapContainer className={"map"} center={defaultPosition} zoom={defaultZoom} zoomControl={false}>
```
Comment (akichula) [inline crap-advisor-frontend/src/core/components/Map/index.jsx:23]: Do not write className prop in curly brakes

File: crap-advisor-frontend/src/core/components/Map/index.jsx [hunk 4]
```diff
@@ -1,29 +1,39 @@
...
+            <TileLayer
+                attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+                url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+            />
+            <ZoomControl position="bottomright"/>
+            {
+                restrooms.map((item, i) => {
+                    return (
+                        <RestroomMarker key={i} props={item}/>
```
Comment (akichula) [inline crap-advisor-frontend/src/core/components/Map/index.jsx:32]: Do not use "props" as a prop name for Component. Need more informative prop name, like “restroom” or other

File: crap-advisor-frontend/src/core/components/RestroomMarker/index.css [hunk 5]
```diff
@@ -0,0 +1,44 @@
+.marker-container {
+    position: relative;
+    top: -15px;
+    left: -15px;
+    width: fit-content;
```
Comment (akichula) [inline crap-advisor-frontend/src/core/components/RestroomMarker/index.css:5]: Woah, "fit-content"? So ludicrous choise

File: crap-advisor-frontend/src/core/components/Sidebar/index.jsx [hunk 6]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
 import "./index.css"
+import logo from "../../../assets/logo.png"
 
 const Sidebar = () => {
     return (
         <div className={"sidebar"}>
-            <h3 id={"appTitle"}>Crap Advisor</h3>
+            <img className={"logo"} src={logo} alt={"Crap Advisor"}/>
```
Comment (akichula) [inline crap-advisor-frontend/src/core/components/Sidebar/index.jsx:8]: Do not write className prop in curly brakes

File: crap-advisor-frontend/src/core/components/Map/index.css (hunk 1 above)
Comment (aiecel) [reply crap-advisor-frontend/src/core/components/Map/index.css:3]: Sadly min- guys made that the map has become larger than my screen

File: crap-advisor-frontend/src/core/components/Map/index.jsx (hunk 2 above)
Comment (aiecel) [reply crap-advisor-frontend/src/core/components/Map/index.jsx:8]: done

File: crap-advisor-frontend/src/core/components/Map/index.jsx (hunk 3 above)
Comment (aiecel) [reply crap-advisor-frontend/src/core/components/Map/index.jsx:23]: Ok, this is my ide doing that

File: crap-advisor-frontend/src/core/components/Map/index.jsx (hunk 4 above)
Comment (aiecel) [reply crap-advisor-frontend/src/core/components/Map/index.jsx:32]: Yeah i wanted to rename it but thought this is some weird frontend convention bullshit or smthn, renamed to "restroom"

File: crap-advisor-frontend/src/core/components/RestroomMarker/index.css (hunk 5 above)
Comment (aiecel) [reply crap-advisor-frontend/src/core/components/RestroomMarker/index.css:5]: Man i was bruteforcing lexems and this is first that worked :|
what should i use?

File: crap-advisor-frontend/src/core/components/Sidebar/index.jsx (hunk 6 above)
Comment (aiecel) [reply crap-advisor-frontend/src/core/components/Sidebar/index.jsx:8]: gotcha

File: crap-advisor-frontend/src/core/components/Map/index.css (hunk 1 above)
Comment (akichula) [reply crap-advisor-frontend/src/core/components/Map/index.css:3]: Uh, okay, max- guys is good enough. I'll mark this conversation as resolved

File: crap-advisor-frontend/src/core/components/RestroomMarker/index.css (hunk 5 above)
Comment (akichula) [reply crap-advisor-frontend/src/core/components/RestroomMarker/index.css:5]: First, try 100%. If it's doenst work well, you can write correct pixels number.

Comment (aiecel) [reply crap-advisor-frontend/src/core/components/RestroomMarker/index.css]: 100% doesn't work, changed to 40px

**Review by akichula — APPROVED** [review]
Code was reviewed, all conversations was resolved. Merge request approved, now it's can be merged

**Comment by aiecel** [comment]
Thanks for everyone, good that my backend work didn't raise any questions

**Pull request merged**